src/main/java/com/scalar/database/transaction/consensuscommit/CommitHandler.java

```diff
@@ -18,21 +18,41 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** */
+/** Handler for the commit phase of a transaction. */
 @ThreadSafe
 public class CommitHandler {
   private static final Logger LOGGER = LoggerFactory.getLogger(CommitHandler.class);
   private final DistributedStorage storage;
   private final Coordinator coordinator;
   private final RecoveryHandler recovery;
 
+  /**
+   * Constructs a {@code CommitHandler} with specified {@link DistributedStorage}, {@link
+   * Coordinator}, and {@link RecoveryHandler}
+   *
+   * @param storage a {@link DistributedStorage}
+   * @param coordinator a {@link Coordinator}
+   * @param recovery a {@link RecoveryHandler}
+   */
   public CommitHandler(
       DistributedStorage storage, Coordinator coordinator, RecoveryHandler recovery) {
     this.storage = checkNotNull(storage);
     this.coordinator = checkNotNull(coordinator);
     this.recovery = checkNotNull(recovery);
   }
 
+  /**
+   * Implements the commit phase of a transaction. That is, all the records that have been changed
+   * by the transaction (dirty objects) will be written to persistent storage.
+   *
+   * <p>The commit phase is performed in two stages: the prepare stage and the commit stage. If the
```

Comment:
No offence at all, but I think this explanation is too vague and seems not really correct.
What is the definition of prepare stage and commit stage here ? And what is the `phase` ?
The code has some methods called `prepareRecords`, `commitState` and `commitRecords` and what belongs to which stage/phase ?
In other words, what do you exactly mean by `If the commit stage finishes successfully` ?